woodwork/type_sys/inference_functions.py

```diff
@@ -1,5 +1,3 @@
-from typing import Callable
-
```

Comment:
Since we're keeping the `InferWithRegex` class the same, no need to delete this annotation.